tests/integration/components/item-component-test.js

```diff
@@ -0,0 +1,54 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, find, render } from '@ember/test-helpers';
+import { run } from '@ember/runloop';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+module('Integration | Component | item-component', function(hooks) {
+  setupRenderingTest(hooks);
+
+  const TEMPLATE = hbs`{{item-component
+    item=item
+    markAsRead=(action markAsRead item)
+  }}`;
+
+  let createItem = function(owner, data) {
+    let store = owner.lookup('service:store');
+    return run(function() {
+      return store.createRecord('item', data);
+    });
+  };
+
+  let setUpAndRender = function(testContext, properties) {
+    testContext.set('item', properties.item);
+    testContext.set('markAsRead', properties.markAsRead || function() {});
+
+    return render(TEMPLATE);
+  };
+
+  test('it renders the content of the item', async function(assert) {
+    const item = createItem(this.owner, {
+      title: 'Hello world'
+    });
+
+    await setUpAndRender(this, { item });
+
+    assert.equal(find('[data-test-item-title]').innerText, item.get('title'));
+  });
+
+  test('it triggers markAsRead action with item when hide button is clicked', async function(assert){
+    const item = createItem(this.owner);
+    const markAsRead = sinon.spy();
```

Comment:
There's another way of doing this without introducing sinon.  You could try something like:
```
const markAsRead = () => assert.ok(true);
```
YMMV :-)

Reply:
I'm so used to have the assert by the end of the scope for readability that it's quite easy to overlook this kind of optimization.